src/client/cmd/start.cpp

```diff
@@ -46,32 +57,58 @@ mp::ReturnCode cmd::Start::run(mp::ArgParser* parser)
         return ReturnCode::Ok;
     };
 
-    auto on_failure = [this, &spinner](grpc::Status& status) {
+    auto on_failure = [this, &spinner, parser](grpc::Status& status) {
         spinner.stop();
-        auto ret = standard_failure_handler_for(name(), cerr, status);
+
+        std::string details;
         if (status.error_code() == grpc::StatusCode::ABORTED && !status.error_details().empty())
         {
             mp::StartError start_error;
             start_error.ParseFromString(status.error_details());
 
-            if (start_error.error_code() == mp::StartError::INSTANCE_DELETED)
+            for (const auto& pair : start_error.instance_errors())
             {
-                fmt::print(cerr, "Use 'recover' to recover the deleted instance or 'purge' to permanently delete the "
-                                 "instance.\n");
+                const auto* err_fmt = unknown_error_fmt;
+                if (pair.second == mp::StartError::INSTANCE_DELETED)
+                    err_fmt = deleted_error_fmt;
+                else if (pair.second == mp::StartError::DOES_NOT_EXIST)
+                {
+                    if (pair.first != petenv_name)
+                        err_fmt = absent_error_fmt;
+                    else
+                        continue;
+                }
+
+                fmt::format_to(std::back_inserter(details), err_fmt, pair.first);
+            }
+
+            if (details.empty())
+            {
+                assert(start_error.instance_errors_size() == 1 &&
+                       std::cbegin(start_error.instance_errors())->first == petenv_name);
+                return run_cmd_and_retry({"multipass", "launch", "--name", petenv_name}, parser, cout, cerr); /*
```

Comment:
Nope, I did it, but clang-format was fine with it. This is an instance of tying the comment to the line of code (following [this suggestion](https://akrzemi1.wordpress.com/2015/04/22/a-clever-comment-style/]). I think it is easier to read than having the function call parameters split across lines to make room for the comment on the same line. Separating the comment just because of formatting would be worse IMO, for the reasons in that article. Let me know...

Reply:
I don't care enough to change it, was just curious if clang-format approved